src/client/article.ts

```diff
@@ -110,6 +110,92 @@ function insertEpic(): void {
     }
 }
 
+interface FormData {
+    [key: string]: string;
+}
+
+function submit(body: FormData, form: Element): void {
+    fetch('https://callouts.code.dev-guardianapis.com/formstack-campaign/submit', {
+        method: 'POST',
+        body: JSON.stringify(body)
+    })
+    .then(() => {
+        form.innerHTML = "<p>Thank you for your contribution</p>"
+    })
+    .catch(() => {
+        const errorPlaceholder = form.querySelector('.error-message');
+        if (errorPlaceholder) {
+            errorPlaceholder.innerHTML = "Sorry, there was a problem submitting your form. Please try again later."
+        }
+    })
+}
+
+function readFile(file: Blob): Promise<string> {
+    return new Promise((resolve, reject) => {
+        const reader = new FileReader();
+        setTimeout(reject, 30000);
+
+        reader.addEventListener('load', () => {
+                if (reader.result) {
+                    const fileAsBase64 = reader.result
+                        .toString()
+                        .split(';base64,')[1];
+                    resolve(fileAsBase64);
+                }
+            }
+        );
+
+        reader.addEventListener('error', () => {
+            reject();
+        });
+
+        reader.readAsDataURL(file);
+    })
+}
+
+function callouts(): void {
+    const callouts = Array.from(document.querySelectorAll('details.callout'));
+    callouts.forEach(callout => {
+        const buttons = Array.from(callout.querySelectorAll('.callout-expand'));
+        buttons.forEach(button => {
+            button.addEventListener('click', () => {
+                callout.toggleAttribute('open');
+            })
+        })
+
+        const form = callout.querySelector('form');
+        if (!form) return;
+        // eslint-disable-next-line @typescript-eslint/no-misused-promises
+        form.addEventListener('submit', async (e): Promise<void> => {
+            try {
+                e.preventDefault();
+                const elements = form.elements as HTMLCollectionOf<HTMLInputElement>;
```

Comment:
Can you avoid the cast like this?

```suggestion
                const elements = form.getElementsByTagName('input');
```